lib/index.d.ts

```diff
@@ -109,20 +117,26 @@ declare namespace yar {
         /**
          * - assigns a value (string, object, etc) to a given key which will persist across requests. Returns the value.
          */
-        set<T>(key: string, value: T): T;
+        set<T extends YarValKeys>(key: T, value: YarValues[T]): YarValues[T];
+        set<T = unknown>(key: string, value: T): T;
```

Comment:
The PR looks all right, why did you need all those declarations with unknown though?

Reply:
@Marsup this allowed the autocomplete to pick up `<T extends YarValKeys>(key: T, value: YarValues[T])` as well as `(key: string, value: T)`, allowing for typed yar values, and also unknown values. Without the `<T = unknown>`, it wouldn't pick up any types and set values to `any`. This was the "AHAH!" moment from @kanongil 's comment.

Reply:
OK, I tend to avoid overloads whenever possible as it makes manipulating types much harder, if there's no other way 🤷🏻‍♂️